new/property.md

```diff
@@ -0,0 +1,206 @@
+
+```ruby
+class MyResource < Chef::Resource
+  property :path, Path, name_attribute: true
+  property :content, String
+end
+```
+
+This calls MyResource.properties[name] = <property type>.
+
+When a property is defined this way:
+- `my_resource.<name>` will get the property value.
+- `my_resource.<name> = <value>` will set the property value.
+- `my_resource.<name> <value>` will set the property value.
+- `MyResource.state_attrs` will include the property name.
+- `MyResource.properties` will contain the property type.
+- The setter and getter manipulate the class variable `@<name>`.
+
+#### Use `property` instead of `attribute` in documentation
+
+`attribute` will continue to be supported; there are simply too many things in the world to deprecate it. However, any generic documentation that talks about attributes will be renamed to talk about properties. `attribute` itself will still be documented.
+
+### Setting properties
+
+### nil
+
+In order to allow for `nil` as a potential explicit value, property setters accept `nil` and set the value of the property to `nil`.  This differs from `attribute`, which considers `property(nil)` to be a get.
+
+#### lazy values
+
+Properties may be set to lazy values, which work the same as in attributes: they are treated as computed values, popped open and validated each time you access them.
+
+```ruby
+file '/x.txt' do
+  content lazy { IO.read("/otherfile.txt") }
+end
+```
+
+### Validation
+
+There are a number of validation parameters to `property` that affect its behavior. If multiple of these are specified, they must *all* succeed.
+
+#### callbacks, kind_of, respond_to, cannot_be, regex, equal_to, required
+
+These function identically to the equivalent functionality on `attribute`.
+
+##### RSpec matchers
+
+`must` allows for rspec 3 matchers to be passed, and will validate them and print failures.
+
+```ruby
+include RSpec::Matchers
+property :path, String, must: start_with('/')
+```
+
+#### must_match
+
+This new option takes an array of values which use Ruby's universal matching
+operator, `===`.  This means that you can type this:
+
+```ruby
+property :x, must_match: [String, :a, :b, nil]
+```
+
+### Other options
+
+#### coerce
+
+`coerce` is a proc run when the user sets a non-lazy value, or reads a lazy or default value. It allows normalization of input, which makes it simple to create expressive interfaces while preserving a simple programming model that knows what to expect:
+
+```ruby
+class File < Chef::Resource
+  attribute :mode, coerce: proc { |v| mode.is_a?(String) ? mode.to_s(8) : mode }
+end
+```
+
+`coerce` procs are run in the context of the instance, so that they have access to other attributes and methods.
+
+#### default
+
+This works similarly to `attribute`, except that `lazy` values are automatically run in the context of the *instance*:
```

Comment:
I think this should be a strict alias. Otherwise we are back to the two systems problem. That means we need to be more careful with compat in the design, but that's okay.